src/Sulu/Bundle/AdminBundle/Resources/js/components/BlockCollection/BlockCollection.js

```diff
@@ -37,14 +41,27 @@ class BlockCollection<T: string, U: {type: T}> extends React.Component<Props<T,
         value: [],
     };
 
+    @observable pasteableBlocks: Array<number> = [];
     @observable generatedBlockIds: Array<number> = [];
     @observable expandedBlocks: Array<boolean> = [];
 
+    fillArraysDisposer: ?() => *;
+    setPasteableBlocksDisposer: ?() => *;
+
     constructor(props: Props<T, U>) {
         super(props);
 
-        this.fillArrays();
-        reaction(() => this.props.value.length, this.fillArrays);
+        this.fillArraysDisposer = reaction(() => this.props.value.length, this.fillArrays, {fireImmediately: true});
+        this.setPasteableBlocksDisposer = clipboardStore.observe(BLOCKS_CLIPBOARD_KEY, action((blocks) => {
+            this.pasteableBlocks = (blocks: any) || [];
+        }));
+
+        this.pasteableBlocks = (clipboardStore.get(BLOCKS_CLIPBOARD_KEY): any) || [];
```

Comment:
Why having a `get` here? I would in `observe` call the callback or was there any problem with `action` inside a constructor?

Reply:
i want to get the current value initially because it might have been copied into the clipboard before the component was mounted and the observer was registered

Reply:
we could change the semantics of observe to fire once when the observer is registered or add a option like `fireImmediately` of mobx 🙂